hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java

```diff
@@ -234,18 +235,22 @@ public EvictionStrategyType getEvictionStrategyType() {
         return EvictionStrategyType.DEFAULT_EVICTION_STRATEGY;
     }
 
-    @Override
+    /**
+     * @deprecated since 3.9, use {@link #getEvictionPolicy()} instead
+     */
+    @Deprecated
     public EvictionPolicyType getEvictionPolicyType() {
-        if (evictionPolicy == EvictionPolicy.LFU) {
-            return EvictionPolicyType.LFU;
-        } else if (evictionPolicy == EvictionPolicy.LRU) {
-            return EvictionPolicyType.LRU;
-        } else if (evictionPolicy == EvictionPolicy.RANDOM) {
-            return EvictionPolicyType.RANDOM;
-        } else if (evictionPolicy == EvictionPolicy.NONE) {
-            return EvictionPolicyType.NONE;
-        } else {
-            return null;
+        switch (evictionPolicy) {
+            case LFU:
+                return EvictionPolicyType.LFU;
+            case LRU:
+                return EvictionPolicyType.LRU;
+            case RANDOM:
+                return EvictionPolicyType.RANDOM;
+            case NONE:
+                return EvictionPolicyType.NONE;
+            default:
+                throw new IllegalArgumentException("Unknown EvictionPolicy configured! " + evictionPolicy);
```

Comment:
I reverted it, just to be sure. We just use it in the `ConfigCompatibilityChecker.MapConfigChecker.check()`, where a `null` value won't hurt even if it would occur.